oops/oops.go

```diff
@@ -171,16 +182,19 @@ func (e *oopsError) writeStackTrace(w io.Writer) {
 		if i > 0 {
 			fmt.Fprintf(w, "\n")
 		}
+		writeFrameTrace(w, stack)
+	}
+}
 
-		for _, frame := range stack {
-			// Print the current function.
-			if frame.Reason != "" {
-				fmt.Fprintf(w, "%s: %s\n", frame.Function, frame.Reason)
-			} else {
-				fmt.Fprintf(w, "%s\n", frame.Function)
-			}
-			fmt.Fprintf(w, "\t%s:%d\n", frame.File, frame.Line)
+func writeFrameTrace(w io.Writer, frames []Frame) {
```

Comment:
nit: it would be good to mention that we are refactoring some part of `writeStackTrace` into a separate function so we can call it from a new method to print the topN frames of the stack etc

Reply:
Done